org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/folding/FoldingRange.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+ * Copyright (c) 2021 TypeFox GmbH (http://www.typefox.io) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.editor.folding;
+
+/**
+ * @author Mark Sujew - Initial contribution and API
+ * 
+ * @since 2.25
+ */
+public class FoldingRange {
+
+	private int offset;
+	private int length;
+	private String kind;
```

Comment:
If we agree on moving this up as API, please add a few words of JavaDoc, e.g. what is the purpose of `kind`. Should the constants from FoldingRangeKind be moved here to avoid the otherwise useless type FoldingRangeKind? Or should kind be of type FoldingRangeKind and that one be a wrapper around String?